File: Java/0455-assign-cookies/0455-assign-cookies.java
public class Solution {
    public int findContentChildren(int[] g, int[] s) {
        // Sort arrays to make the matching process efficient
        Arrays.sort(g);
        Arrays.sort(s);

        int count = 0;
        int cookieIndex = 0;

        // Iterate through each child's greed factor
        for (int i = 0; i < g.length; i++) {
            // Try to find a suitable cookie for the current child
            while (cookieIndex < s.length && s[cookieIndex] < g[i]) {
                cookieIndex++;
            }

            // If a suitable cookie is found, increment count
            if (cookieIndex < s.length) {
                count++;
                cookieIndex++;
            }
        }

        return count;
    }
}

File: C++/0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
    public:
        int findContentChildren(vector<int>& g, vector<int>& s) {
            int ans = 0;
    
            // Check if either array is empty
            if (g.empty() || s.empty()) {
                return 0;
            }
    
            // Sort arrays to make the matching process efficient
            sort(g.begin(), g.end());
            sort(s.begin(), s.end());
    
            int gi = 0;
            int si = 0;
    
            // Iterate through each child's greed factor
            for (int child : g) {
                // Try to find a suitable cookie for the current child
                while (si < s.size() && s[si] < child) {
                    si++;
    
                    // All cookies are content
                    if (si >= s.size()) {
                        return ans;
                    }
                }
    
                si++;
                ans++;
    
                // All cookies are content
                if (si >= s.size()) {
                    return ans;
                }
            }
    
            return ans;
        }
    };